shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1727,10 +1727,13 @@ void WebContents::EnableDeviceEmulation(
   if (frame_host) {
     auto* widget_host =
         frame_host ? frame_host->GetView()->GetRenderWidgetHost() : nullptr;
-    if (!widget_host)
-      return;
-    widget_host->Send(new WidgetMsg_EnableDeviceEmulation(
-        widget_host->GetRoutingID(), params));
+    if (widget_host) {
+      auto* impl = content::RenderWidgetHostImpl::FromID(
```

Comment:
`RenderWidgetHostImpl` is implementation of `RenderWidgetHost` class, you can static cast safely.

```
auto* rwh_impl =
   frame_host ?
   static_cast<content::RenderWidgetHostImpl*>(frame_host->GetView()->GetRenderWidgetHost())
   : nullptr;

if (rwh_impl) {

}
```